File: app/in/partake/controller/api/admin/AdminEventIndexRecreationAPI.java
package in.partake.controller.api.admin;

import in.partake.app.PartakeApp;
import in.partake.base.PartakeException;
import in.partake.controller.api.AbstractPartakeAPI;
import in.partake.model.IPartakeDAOs;
import in.partake.model.access.Transaction;
import in.partake.model.dao.DAOException;
import in.partake.model.dao.PartakeConnection;
import in.partake.model.daofacade.EventDAOFacade;
import in.partake.service.IEventSearchService;
import play.mvc.Result;

public class AdminEventIndexRecreationAPI extends AbstractPartakeAPI {

    public static Result post() throws DAOException, PartakeException {
        return new AdminEventIndexRecreationAPI().execute();
    }

    public Result doExecute() throws DAOException, PartakeException {
        ensureAdmin();
        ensureValidSessionToken();

        new EventIndexRecreationTransaction().execute();

        return renderOK();
    }
}

class EventIndexRecreationTransaction extends Transaction<Void> {

    @Override
    protected Void doExecute(PartakeConnection con, IPartakeDAOs daos) throws DAOException, PartakeException {
        IEventSearchService searchService = PartakeApp.getEventSearchService();
        EventDAOFacade.recreateEventIndex(con, daos, searchService);
        return null;
    }
}

File: app/controllers/api/admin/AdminEventIndexRecreationAPI.scala
package controllers.api.admin

import in.partake.app.PartakeApp
import in.partake.base.PartakeException
import in.partake.model.IPartakeDAOs
import in.partake.model.dao.DAOException
import in.partake.model.dao.PartakeConnection
import in.partake.model.daofacade.EventDAOFacade
import in.partake.service.IEventSearchService

import controllers.api.AbstractAPI
import controllers.ActionContext
import models.db.Transaction

import play.api.libs.json.JsObject
import play.api.libs.json.Json
import play.api.mvc.Request
import play.api.mvc.AnyContent
import play.api.mvc.PlainResult
import play.api.mvc.Result

object AdminEventIndexRecreationAPI extends AbstractAPI[Unit, Unit] {
  override def parseRequest(request: Request[AnyContent])(implicit context: ActionContext): Unit = {
    ensureAdmin()
    ensureValidSessionToken(request)
  }

  override def executeAction(params: Unit)(implicit context: ActionContext): Unit = {
    Transaction { (con, daos) =>
      val searchService: IEventSearchService = PartakeApp.getEventSearchService()
      EventDAOFacade.recreateEventIndex(con, daos, searchService)
    }
  }

  override def renderResult(values: Unit)(implicit context: ActionContext): PlainResult = {
    renderOK()
  }
}

